cluster_management/src/main/java/com/pinterest/rocksplicator/task/RestoreTask.java

```diff
@@ -129,13 +130,24 @@ protected void executeRestore(String host, int adminPort, String dbName, String
                                 boolean useS3Store, String s3Bucket)
       throws RuntimeException {
     try {
+      CheckDBResponse
+          checkDBResponse =
+          Utils.checkRemoteOrLocalDB(host, adminPort, dbName, false, null, null);
+      boolean isLeader = checkDBResponse.isIs_master();
```

Comment:
Is it possible that a leader can change (from a helix POV) while the restoring is taking place?

Reply:
Simple Restore is basically overwriting any existing data on disk, thus, it does not matter leader change or not. 
the best practice to use it will be enable maintenance mode first before restore.

Reply:
@rajathprasad any concern even if we have maintenance model on?

Reply:
But we cannot enforce that, right?